bundles/org.openhab.core.audio/src/main/java/org/openhab/core/audio/AudioSink.java

```diff
@@ -58,13 +58,45 @@ public interface AudioSink {
      *
      * In case the audioStream is null, this should be interpreted as a request to end any currently playing stream.
      *
+     * When the process method ends, if the stream implements the {@link org.openhab.core.common.Disposable} interface,
+     * the sink should hereafter get rid of it by calling the dispose method.
+     *
      * @param audioStream the audio stream to play or null to keep quiet
      * @throws UnsupportedAudioFormatException If audioStream format is not supported
      * @throws UnsupportedAudioStreamException If audioStream is not supported
      */
     void process(@Nullable AudioStream audioStream)
             throws UnsupportedAudioFormatException, UnsupportedAudioStreamException;
 
+    /**
+     * Processes the passed {@link AudioStream}, and executes the whenFinished Runnable (or stores for later execution
+     * if the sink is asynchronous).
+     *
+     * If the passed {@link AudioStream} is not supported by this instance, an {@link UnsupportedAudioStreamException}
+     * is thrown.
+     *
+     * If the passed {@link AudioStream} has an {@link AudioFormat} not supported by this instance,
+     * an {@link UnsupportedAudioFormatException} is thrown.
+     *
+     * In case the audioStream is null, this should be interpreted as a request to end any currently playing stream.
+     *
+     * When the process method ends, if the stream implements the {@link org.openhab.core.common.Disposable} interface,
+     * the sink should hereafter get rid of it by calling the dispose method.
+     *
+     * @param audioStream the audio stream to play or null to keep quiet
+     * @param whenFinished A Runnable to run when the sound is finished playing.
+     * @throws UnsupportedAudioFormatException If audioStream format is not supported
+     * @throws UnsupportedAudioStreamException If audioStream is not supported
+     */
+    default void process(@Nullable AudioStream audioStream, Runnable whenFinished)
+            throws UnsupportedAudioFormatException, UnsupportedAudioStreamException {
+        try {
+            process(audioStream);
+        } finally {
+            whenFinished.run();
+        }
+    }
```

Comment:
You assume a synchronous behaviour by default here. But as you are not handling Closable (deleting source) here, that should be OK, meaning no change compared to the current behaviour.

Reply:
Yes, exactly, no change.

> Asynchronous audio sinks will have to move to AudioSinkAsync to have the volume restored after the stream is fully played.

Yes, sinks that choose to extends AudioSinkAsync are then responsible to execute the Runnable.  To get it, they have to either overriding the process with Runnable arg, or use the runDelayed method that stores it for them.